lib/cli/cliCert.js

```diff
@@ -0,0 +1,139 @@
+'use strict';
+const CLICommand = require('./cliCommand.js');
+const CLIObjects = require('./cliObjects.js');
+const messages = require('./messages.js');
+const tools = require('../tools.js');
+const forge = require('node-forge');
+const os = require('os');
+
+/** Command ioBroker cert ... */
+module.exports = class CLICert extends CLICommand {
+
+    /** @param {import('./cliCommand.js').CLICommandOptions} options */
+    constructor(options) {
+        super(options);
+    }
+
+    /**
+     * Executes a command
+     * @param {any[]} args
+     */
+    execute(args) {
+        const { callback, showHelp } = this.options;
+        const command = args[0];
+
+        switch (command) {
+            case 'renew':
+            case 'create':
+                return this.create(args);
+            default:
+                messages.error.unknownCommand('cert', command);
+                showHelp();
+                return void callback(3);
+        }
+    }
+
+    /**
+     * create new private certificate
+     * @param {any[]} args
+     */
+    create(args) {
+
+        // If at any time you wish to disable the use of native code, where available, for particular forge features
+        // like its secure random number generator, you may set the forge.options.usePureJavaScript flag to true. It
+        // is not recommended that you set this flag as native code is typically more performant and may have stronger
+        // security properties. It may be useful to set this flag to test certain features that you plan to run in
+        // environments that are different from your testing environment.
+        forge.options.usePureJavaScript = false;
+
+        const { callback, dbConnect } = this.options;
+        const id = "system.certificates";
+        const certPropPath = "native.certificates";
+
+        // https://github.com/digitalbazaar/forge
+        const pki = forge.pki;
+
+        const keys = pki.rsa.generateKeyPair({bits: 2048, e: 0x10001});
+        let cert = pki.createCertificate();
+
+        cert.publicKey = keys.publicKey;
+        cert.serialNumber = '01';
+        cert.validity.notBefore = new Date();
+        cert.validity.notAfter = new Date();
+        cert.validity.notAfter.setFullYear(cert.validity.notBefore.getFullYear() + 5);
+
+        const subAttrs = [
+            {name:'commonName',value: tools.getHostName()},
+            {name:'organizationName',value:'ioBroker GmbH'},
+            {shortName:'OU',value:'iobroker'}
+        ];
+
+        const issAttrs = [
+            {name:'commonName',value: 'iobroker'},
+            {name:'organizationName',value:'ioBroker GmbH'},
+            {shortName:'OU',value:'iobroker'}
```

Comment:
What does OU mean?

Reply:
In computing, an organizational unit (OU) provides a way of classifying objects located in directories, or names in a digital certificate hierarchy, typically used either to differentiate between objects with the same name (John Doe in OU "marketing" versus John Doe in OU "customer service"), or to parcel out authority to create and manage objects (for example: to give rights for user-creation to local technicians instead of having to manage all accounts from a single central group).
The name organizational unit appears to represent a single organization with multiple units (departments) within that organization. However, OUs do not always follow this model. They might represent geographical regions, job-functions, associations with other (external) groups, or the technology used in relation to the objects.
In this case the OU represents the product that is provided by ioBroker GmbH.